File: app-mvc-kotlin/src/main/java/com/naveed/samples/data/network/IMoviesApi.java
package com.naveed.samples.data.network;


import com.naveed.samples.data.models.ListMoviesResp;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface IMoviesApi {


    @GET("movie/{type}")
    Call<ListMoviesResp> getMoviesList(@Path("type") String type,
                                       @Query("api_key") String apiKey,
                                       @Query("language") String language,
                                       @Query("page") int page,
                                       @Query("region") String region);

    @GET("search/movie")
    Call<ListMoviesResp> getSearchAbleResultsMoviesList(@Query("api_key") String apiKey,
                                       @Query("query") String language,
                                       @Query("page") int page);



}

File: app/src/main/java/com/naveed/samples/data/network/IMoviesApi.kt
package com.naveed.samples.data.network


import com.naveed.samples.data.models.ListMoviesResp
import io.reactivex.Observable
import io.reactivex.Single
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface IMoviesApi {


    @GET("movie/{type}")
    fun getMoviesList(@Path("type") type: String,
                      @Query("api_key") apiKey: String,
                      @Query("language") language: String,
                      @Query("page") page: Int,
                      @Query("region") region: String): Observable<ListMoviesResp>

    @GET("search/movie")
    fun getSearchAbleResultsMoviesList(@Query("api_key") apiKey: String,
                                       @Query("query") language: String,
                                       @Query("page") page: Int): Single<ListMoviesResp>


}


